### PR TITLE
test: remove deepspeed marks from dsat tests

### PR DESCRIPTION
**Pull request by garrett361**
## Description

The `dsat` nightly distributed tests had incorrect `@pytest.mark.deepspeed` marks, in addition to `@pytest.mark.distributed` marks. They should only have the latter marks.  This PR removes the `@pytest.mark.deepspeed` marks.

## Test Plan
Verify on CircleCI that the modified `dsat` tests are only run in one workflow: nightly `test-e2e-gpu-distributed`

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by tayritenour — COMMENTED** [review]
Are we changing the semantics of the `pytest.mark.deepspeed` somehow? I thought it was for examples that need the deepspeed library in the container?

File: e2e_tests/tests/nightly/test_distributed.py [hunk 1]
```diff
@@ -344,7 +344,6 @@ def test_textual_inversion_stable_diffusion_generate() -> None:
 
 @pytest.mark.distributed
 @pytest.mark.gpu_required
-@pytest.mark.deepspeed
```
Comment (wes-turner) [inline e2e_tests/tests/nightly/test_distributed.py:347]: What makes a test "deepspeed" vs. not? Being a dummy, I see the word "deepspeed" and think that makes it a member of the club.

**Review by wes-turner — APPROVED** [review]

File: e2e_tests/tests/nightly/test_distributed.py (hunk 1 above)
Comment (wes-turner) [reply e2e_tests/tests/nightly/test_distributed.py:347]: Maybe relevant for future refactoring, what ties the tests in `test_distributed.py` together? Why do they all share the same file?

**Comment by garrett361** [comment]
> Are we changing the semantics of the `pytest.mark.deepspeed` somehow? I thought it was for examples that need the deepspeed library in the container?

@tayritenour I thought so, too, but it was actually just causing the test to redundantly run in two separate workflows: [test-e2e-gpu-distributed](https://app.circleci.com/pipelines/gh/determined-ai/determined/44291/workflows/ba96ce4c-64bd-492c-9506-58d99287a4ac/jobs/1795361/parallel-runs/0/steps/0-129) and [test-e2e-gpu-deepspeed](https://app.circleci.com/pipelines/gh/determined-ai/determined/44291/workflows/ba96ce4c-64bd-492c-9506-58d99287a4ac/jobs/1795362/parallel-runs/0/steps/0-129)

File: e2e_tests/tests/nightly/test_distributed.py (hunk 1 above)
Comment (garrett361) [reply e2e_tests/tests/nightly/test_distributed.py:347]: >What makes a test "deepspeed" vs. not? Being a dummy, I see the word "deepspeed" and think that makes it a member of the club.

yeah, that was part of my confusion and why I thought the mark was needed, but AFAICT it just marks the tests to use slightly different compute resources.

**Review by tayritenour — APPROVED** [review]
If it's just causing the tests to be redundant, then it makes sense to me

File: e2e_tests/tests/nightly/test_distributed.py (hunk 1 above)
Comment (garrett361) [reply e2e_tests/tests/nightly/test_distributed.py:347]: >Maybe relevant for future refactoring, what ties the tests in test_distributed.py together? Why do they all share the same file?

Should just be that all the tests require more than one GPU, I believe

**Pull request merged**